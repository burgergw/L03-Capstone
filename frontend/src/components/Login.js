//Importing the required packages
import React from 'react'
import { useState } from 'react';
//React functional arrow component to handle user login
const Login = () => {
    //Creating state variables
    const [logUsername, setLogUsername] = useState('');
    const [logPassword, setLogPassword] = useState('');
    const [logAdmin, setLogAdmin] = useState();

    const login = async (e) => {
        //Function to handle user login
        e.preventDefault();
        //Post request to send the login info to the backend
        const response = await fetch('http://localhost:5150/login', {
            method: "POST",
            headers: {"Content-Type": "application/json"},
            body: JSON.stringify({
                username: logUsername,
                password: logPassword
            })
        })
        //Creating object from response
        const data = await response.json()
        //Check if a token is returned
        if (data.token) {
            //Storing the returned token in local storage
            localStorage.setItem('token', data.token);
            //Navigating user to the display page when logged in
            window.location.href = '/display';
          //If no token is returned, login will be unsuccessful  
        } else {
            console.log('Login Unsuccessful')
        }
        
    }
  //Rendering the login form  
  return (
    <div>
        <h1>Login</h1>
        <form onSubmit={login}>
            <input type='username' placeholder='Enter username...' value={logUsername} onChange={e => setLogUsername(e.target.value)}/>
            <input type='password' placeholder='Enter password...' value={logPassword} onChange={e => setLogPassword(e.target.value)}/>
            <button type='submit'>Login</button>
        </form>
    </div>
  )
}
//Exporting functional component
export default Login;