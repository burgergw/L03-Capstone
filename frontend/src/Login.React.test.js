import React from "react";
import Login from "./components/Login";
import renderer from 'react-test-renderer';

test('Login component renders correctly', () => {
    const tree = renderer
    .create(<Login/>).toJSON();
    expect(tree).toMatchSnapshot();
})