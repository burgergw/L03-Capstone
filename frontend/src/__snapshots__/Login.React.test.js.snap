// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<div>
  <h1>
    Login
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Enter username..."
      type="username"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Enter password..."
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
